src/server/authOptions.ts

```diff
@@ -8,33 +8,50 @@ import GithubProvider from 'next-auth/providers/github'
 import EmailProvider, {EmailUserConfig} from 'next-auth/providers/email'
 import CredentialsProvider from 'next-auth/providers/credentials'
 import {DefaultJWT} from 'next-auth/jwt'
+import {z} from 'zod'
 
 import {verifyCredentials} from './actions/auth/verifyCredentials'
 import {UserRole} from './db/models'
 
-import {env} from '@/env'
 import {db} from '@/server/db'
+import {env} from '@/env'
 
 declare module 'next-auth' {
   interface Session extends DefaultSession {
     user: {
       id: string
-      role: UserRole
+      role: UserRole | null
       provider: string | undefined
     } & DefaultSession['user']
   }
 }
 
 declare module 'next-auth/jwt' {
   interface JWT extends DefaultJWT {
-    role: UserRole
+    role: UserRole | null
     provider: string | undefined
   }
 }
 
+const updateSessionSchema = z
+  .object({
+    role: z.enum([UserRole.AUDITOR, UserRole.PROJECT_OWNER]),
+  })
+  .partial()
+  .optional()
+
 export const authOptions: NextAuthOptions = {
+  pages: {
+    signIn: '/auth/signin',
+  },
   callbacks: {
-    jwt: async ({token, user, trigger, account}) => {
+    jwt: async ({token, user, trigger, account, session}) => {
+      const sessionData = updateSessionSchema.parse(session)
+
+      if (trigger === 'update' && sessionData?.role) {
+        token.role = sessionData.role
+      }
```

Comment:
Eventually yes, but since we need the role data to be updated immediately we use the `session.update()` method. This is the `session`  object passed to the `jwt` callback so we can update the token appropriately